src/picongpu/include/fields/currentInterpolation/NoneDS/NoneDS.hpp

```diff
@@ -0,0 +1,177 @@
+     *       with AssignIfInRange...
+     *
+     * \tparam T_simDim maximum dimensionality of the mesh
+     * \tparam T_direction (0)X (1)Y or (2)Z for the direction one wants to
+     *                     shift to
+     * \tparam isShiftAble auto-filled value that decides if this direction
+     *                     is actually non-existent == periodic
+     */
+    template<uint32_t T_simDim, uint32_t T_direction, bool isShiftAble=(T_direction<T_simDim) >
+    struct ShiftMeIfYouCan
+    {
+        static const uint32_t dim = T_simDim;
+        static const uint32_t dir = T_direction;
+
+        template<class T_DataBox >
+        HDINLINE T_DataBox operator()(const T_DataBox& dataBox) const
+        {
+            DataSpace<dim> shift;
+            shift[dir] = 1;
+            return dataBox.shift(shift);
+        }
+    };
+
+    template<uint32_t T_simDim, uint32_t T_direction>
+    struct ShiftMeIfYouCan<T_simDim, T_direction, false>
+    {
+        template<class T_DataBox >
+        HDINLINE T_DataBox operator()(const T_DataBox& dataBox) const
+        {
+            return dataBox;
+        }
+    };
+
+    /* that is not a "real" yee curl, but it looks a bit like it */
+    template<class Difference>
+    struct ShiftCurl
+    {
+        typedef typename Difference::OffsetOrigin LowerMargin;
+        typedef typename Difference::OffsetEnd UpperMargin;
+
+        template<class DataBox >
+        HDINLINE typename DataBox::ValueType operator()(const DataBox& mem) const
+        {
+            const typename Difference::template GetDifference<0> Dx;
+            const typename Difference::template GetDifference<1> Dy;
+            const typename Difference::template GetDifference<2> Dz;
+
+            const ShiftMeIfYouCan<simDim, 0> sx;
+            const ShiftMeIfYouCan<simDim, 1> sy;
+            const ShiftMeIfYouCan<simDim, 2> sz;
+
+            return float3_X(Dy(sx(mem)).z() - Dz(sx(mem)).y(),
+                            Dz(sy(mem)).x() - Dx(sy(mem)).z(),
+                            Dx(sz(mem)).y() - Dy(sz(mem)).x());
+        }
+    };
+} /* namespace detail */
+
+template<uint32_t T_simDim>
+struct NoneDS
+{
+    static const uint32_t dim = T_simDim;
+
+    typedef typename PMacc::math::CT::make_Int<dim, 0>::type LowerMargin;
+    typedef typename PMacc::math::CT::make_Int<dim, 1>::type UpperMargin;
+
+    template<typename DataBoxE, typename DataBoxB, typename DataBoxJ>
+    HDINLINE void operator()(DataBoxE fieldE,
+                             DataBoxB fieldB,
+                             DataBoxJ fieldJ )
+    {
+        typedef typename DataBoxJ::ValueType TypeJ;
+        typedef typename GetComponentsType<TypeJ>::type ComponentJ;
+
+        const DataSpace<dim> self;
+
+        const ComponentJ deltaT = DELTA_T;
+        const ComponentJ constE = (float_X(1.0) / EPS0) * deltaT;
+        const ComponentJ constB = (float_X(0.5) / EPS0) * deltaT * deltaT;
```

Comment:
according to Sentoku's derivation,<sup>4</sup> this should be `0.25`.
changing this reduces the heating but is still too high (in the few-% level) with TSC & Esirkepov and the KHI still shows artifacts.